util/metric/metric.go

```diff
@@ -205,15 +232,29 @@ func (c *Counter) MarshalJSON() ([]byte, error) {
 	return json.Marshal(c.Counter.Count())
 }
 
+// Desc implements prometheus.Metric.
+func (c *Counter) Desc() *prometheus.Desc {
+	return c.desc
+}
+
+// Write implements prometheus.Metric.
+func (c *Counter) Write(met *prometheus_model.Metric) error {
+	met.Gauge = &prometheus_model.Gauge{Value: proto.Float64(float64(c.Count()))}
```

Comment:
Ah, that might work better, although we are concerned about efficiency as well since we expect quite a few metrics, some will lock (as opposed to simple atomic values), and may be accessed from all over the code.

I'm using Gauge for now since our underlying Counter implementation allows both `Inc` and `Dec`. I was going audit the code a bit to make sure we never decrements `Counter`, in which case I could prohibit `Dec` and switch to the prometheus `Counter`